server/src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java

```diff
@@ -74,6 +74,10 @@ public LeafBucketCollector getLeafCollector(final AggregationExecutionContext ag
         IndexReaderContext topLevelContext = ReaderUtil.getTopLevelContext(aggCtx.getLeafReaderContext());
         IndexSearcher searcher = new IndexSearcher(topLevelContext);
         searcher.setQueryCache(null);
+        if (scoreMode() != ScoreMode.COMPLETE_NO_SCORES) {
```

Comment:
What I like about this approach is that, you have to be specific and opt-out of scoring. Is scripted_metric aggregation the only aggregation we need to added another option? 

I suspect that this if statement will break a lot of users. Also the array out of bounds error only happened in combination with the composite agg (essentially as part of deferred collection). When no composite agg is used then we fail as well, while previously users would be able to run aggregations with nested and a sub agg that requires score. In this case the score being accessible to these sub aggregations is the _score of the root document and not the individual matching nested documents. I suspect for certain users this is ok (same _score for aggs above the nested agg as sub aggs of nested agg), but I'm not sure.

If we would change `Composite#runDeferredCollections()` method (first line/statement) to not needing scores if parent aggregation is a nested aggregation then we would avoid the array out of bounds exception and the emitting of incorrect scores (this only occurs with `Compsite` agg if I recall correctly). I don't think user actually cared about scores in the case. First we emitted the wrong scores any way (scores for the wrong root document) and secondly I think scores were computed because scripted metric agg was used or top hits with default sort. This approach would mean that we would't break in a way that this currently breaks.